src/main/java/eu/bbmri/eric/csit/service/negotiator/mappers/RequestModelsMapper.java

```diff
@@ -79,63 +71,32 @@ void addMappings() {
             mapper
                 .using(queryToRedirectUri)
                 .map(requestDTO -> requestDTO, QueryV2DTO::setRedirectUri));
+
+    TypeMap<RequestCreateDTO, Request> requestCreateDTORequestTypeMap =
+        modelMapper.createTypeMap(RequestCreateDTO.class, Request.class);
   }
 
   private String convertNegotiationToNegotiationId(Negotiation negotiation) {
     return negotiation != null ? negotiation.getId() : null;
   }
 
   private String convertIdToRedirectUrl(String requestId) {
+    if (FRONTEND_URL.endsWith("/")) {
```

Comment:
Instead of checking it each time, you can check it in the constructor and assign the value without the "/"